app/app_test.go

```diff
@@ -118,7 +243,72 @@ func Test_NormalFlow(t *testing.T) {
 		true,
 		srvs,
 		false,
+		nil,
+	)
+}
+
+func Test_NormalFlowHalfHalfSingleRun(t *testing.T) {
+	valAmt := 10
+	operatorAmt := 4
+
+	lock, enrs, keyShares := cluster.NewForT(
+		t,
+		valAmt,
+		operatorAmt,
+		operatorAmt,
+		0,
+		rand.New(rand.NewSource(0)),
+		cluster.WithVersion("v1.8.0"),
+	)
+
+	vs := validatorSetFromLock(lock)
+	var vsLock sync.Mutex
+
+	// set the first 5 as non-active
+	for i := 0; i < 5; i++ {
+		vs[eth2p0.ValidatorIndex(i+1)].Status = eth2v1.ValidatorStatePendingQueued
+	}
+
+	srvs := newServers(t, lock, func() beaconmock.ValidatorSet {
+		vsLock.Lock()
+		defer vsLock.Unlock()
+
+		cl := make(beaconmock.ValidatorSet)
+
+		for _, v := range vs {
+			cl[v.Index] = cloneValidator(v)
+		}
+
+		return cl
+	})
+
+	defer srvs.Close()
+
+	runDone := make(chan struct{})
+
+	//nolint:testifylint // needed for concurrent test
+	go run(t,
```

Comment:
Assertions in separate go routine from the main go routine from which the test is executed, are not properly evaluated. If you check this [example](https://go.dev/play/p/WoBGMiKQDEk) and run it 10 times you will see different result of the test passing or not.

It's better if we send the result to the main routine through channel.